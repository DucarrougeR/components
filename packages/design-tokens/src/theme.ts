import { InterpolationValue } from 'styled-components'
import {
  defaultReset,
  Easings,
  FontFamilyChoices,
  FontSizeRamp,
  FontWeightRamp,
  LineHeightRamp,
  Palette,
  Radii,
  SemanticColors,
  Shadows,
  SpaceRamp,
  Transitions,
} from './system'

/**
 * Theme attributes shouldn't be exported as they should be consumed via `theme` rather than via
 * direct import.
 */
import {
  breakpoints,
  easings,
  fontFamilies,
  fontSizes,
  fontWeights,
  lineHeights,
  palette,
  radii,
  semanticColors,
  shadows,
  space,
  transitions,
} from './tokens'

export interface Theme {
  breakpoints: string[]
  colors: { palette: Palette; semanticColors: SemanticColors }
  easings: Easings
  fontSizes: FontSizeRamp
  fontWeights: FontWeightRamp
  fonts: FontFamilyChoices
  lineHeights: LineHeightRamp
  radii: Radii
  /**
   * A function that can be overridden to return different reset css properties
   * or null to remove all resets. Most base elements in Lens implement the reset.
   */
  reset: () => InterpolationValue
  shadows: Shadows
  space: SpaceRamp
  transitions: Transitions
}

export const theme: Theme = {
  breakpoints,
  colors: { palette, semanticColors },
  easings,
  fontSizes,
  fontWeights,
  fonts: fontFamilies,
  lineHeights,
  radii,
  reset: defaultReset,
  shadows,
  space,
  transitions,
}
