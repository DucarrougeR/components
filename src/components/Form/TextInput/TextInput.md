```js
  <TextInput id="theme-name" name="themeName" label="Theme Name" />
  <TextInput id="font-family" name="themeFontFamly" label="Font Family" />
  <TextInput id="background-color" name="themeBackgroundColor" label="Background Color" />
```
