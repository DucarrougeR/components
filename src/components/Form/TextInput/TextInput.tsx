// Typescript component boilerplate for generating meaningful declaration files.
import * as React from 'react'
import styled, { StyledComponentClass } from '../../../styled_components'
export { StyledComponentClass }
import { ThemeInterface } from '../../../themes'
export { ThemeInterface }
import { themeSpacing } from '../../../themes/theme_spacing'
// End Typescript component boilerplate

import { FormGroup } from '../FormGroup/FormGroup'
import { Input, InputTypes } from '../Input/Input'
import { Label } from '../Label/Label'

export interface TextInputProps {
  id: string
  label: string
  name: string
  placeholder?: string
  required?: boolean
}

/**
 * Headings are used to help users understand  what a major section of an interface is about, for example the labeling
 * of a page or a title of a card component.
 */
const TextInputGenerator: React.SFC<TextInputProps> = ({
  id,
  label,
  name,
  placeholder,
  required,
  ...args
}) => {
  // // This prevents our props from being passed directly to the underlying tags, which ultimately
  // // would cause some warnings. Ideally we would define the return type for this function, and
  // // Typescript would warn us when passing props that are invalid.
  // //
  // // See https://reactjs.org/warnings/unknown-prop.html
  // const props = Object.assign({}, args)
  // delete props.id
  // delete props.label
  // delete props.name
  // delete props.placeholder
  // delete props.required

  return (
    <FormGroup>
      <Label htmlFor={id}>{label}</Label>
      <Input
        id={id}
        type={InputTypes.Text}
        name={name}
        placeholder={placeholder}
        required={required}
        {...args}
      />
    </FormGroup>
  )
}

export const TextInput = styled<TextInputProps>(TextInputGenerator)``
