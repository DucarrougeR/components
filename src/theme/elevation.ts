export type ElevationLayers = '1' | '2' | '3' | '4' | '5' | '6' | '7' | 'max'

export type Shadows = { [K in ElevationLayers]?: string }

export const shadows: Shadows = {
  1: '0 1px 2px rgba(0, 0, 0, 0.1), 0px 1px 1px rgba(0, 0, 0, 0.05)',
  2: '0 3px 4px rgba(0, 0, 0, 0.11), 0 1px 2px rgba(0, 0, 0, 0.04)',
  3: '0 6px 10px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.04)',
  4: '0 9px 18px rgba(0, 0, 0, 0.14), 0 1px 2px rgba(0, 0, 0, 0.04),  0 10px 8px 4px rgba(0, 0, 0, 0.01)',
  5: '0 14px 20px rgba(0, 0, 0,.15), 0 1px 1px rgba(0, 0, 0, 0.04),  0 14px 10px 8px rgba(0, 0, 0, 0.02)',
  6: '0 22px 42px 0px rgba(0, 0, 0,.18),  0 1px 1px rgba(0, 0, 0, 0.05), 0 16px 20px 10px rgba(0, 0, 0, 0.025)',
}
