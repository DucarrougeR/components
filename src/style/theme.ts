import { rem } from 'polished'
import { InterpolationValue } from 'styled-components'
import {
  SizeLarge,
  SizeMedium,
  SizeNone,
  SizeSmall,
  SizeXLarge,
  SizeXSmall,
  SizeXXLarge,
  SizeXXXLarge,
} from '../types'
import { Easings, easings } from './easings'
import { shadows, Shadows } from './elevation'
import { fontFaces, FontFaces } from './font_faces'
import { FontRamp, fontSizes, lineHeights } from './font_sizes'
import { fontWeights } from './font_weights'
import { palette, Palette } from './palette'
import { SemanticColors, semanticColors } from './semantic_colors'
import { transitions, Transitions } from './transitions'

export * from './easings'
export * from './elevation'
export * from './font_sizes'
export * from './palette'
export * from './semantic_colors'
export * from './transitions'

export type SpacingSizes =
  | SizeNone
  | SizeLarge
  | SizeMedium
  | SizeSmall
  | SizeXLarge
  | SizeXSmall
  | SizeXXLarge
  | SizeXXXLarge

export interface Theme {
  breakpoints: string[]
  easings: Easings
  fontFaces: FontFaces
  fontSizes: FontRamp
  fontWeights: Record<string, number>
  lineHeights: FontRamp
  palette: Palette
  /**
   * A function that can be overridden to return different reset css properties
   * or null to remove all resets. Most base elements in Lens implement the reset.
   */
  reset: () => InterpolationValue
  semanticColors: SemanticColors
  shadows: Shadows
  space: Record<SpacingSizes, string>
  transitions: Transitions
}

export const theme: Theme = {
  breakpoints: ['30rem', '48rem', '64rem', '75rem', '90rem'],
  easings,
  fontFaces,
  fontSizes,
  fontWeights,
  lineHeights,
  palette,
  reset: () => {
    return {
      border: 0,
      font: 'inherit',
      fontSize: '100%',
      margin: 0,
      padding: 0,
      verticalAlign: 'baseline',
    }
  },
  semanticColors,
  shadows,
  space: {
    // tslint:disable:object-literal-sort-keys
    none: 'none',
    xsmall: rem(4),
    small: rem(8),
    medium: rem(16),
    large: rem(20),
    xlarge: rem(32),
    xxlarge: rem(40),
    xxxlarge: rem(60),
    // tslint:enable:object-literal-sort-keys
  },
  transitions,
}
