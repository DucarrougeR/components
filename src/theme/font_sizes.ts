import { rem } from 'polished'
import { RampSizes } from '../style/ramp_sizes'

export type FontRamp = { [K in RampSizes]: string }

// tslint:disable:object-literal-sort-keys
export const fontRamp: FontRamp = {
  d1: rem(58),
  d2: rem(46),
  d3: rem(36),
  1: rem(25),
  2: rem(22),
  3: rem(18),
  4: rem(16),
  5: rem(14),
  6: rem(12),
}

// tslint:disable:object-literal-sort-keys
export const lineHeightRamp: FontRamp = {
  d1: rem(84),
  d2: rem(64),
  d3: rem(52),
  1: rem(40),
  2: rem(32),
  3: rem(28),
  4: rem(24),
  5: rem(20),
  6: rem(16),
}
