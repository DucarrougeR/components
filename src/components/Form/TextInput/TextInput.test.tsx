import * as React from 'react'
import { TextInput } from './TextInput'
import { create } from 'react-test-renderer'
import 'jest-styled-components'
import theme from '../../../themes'

test('A TextInput', () => {
  const component = create(
    <TextInput label="🍕🎉" name="pizzaParty" id="pizza-party" theme={theme} />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
