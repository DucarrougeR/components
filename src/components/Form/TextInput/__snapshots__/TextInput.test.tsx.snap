// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A TextInput 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 8px;
}

.c3 {
  border: solid 1px #C8CBCC;
  height: 28px;
  padding: 0 8px;
  border-radius: 4px;
  font-size: 14px;
}

.c1 {
  font-weight: 600;
  color: #1E2933;
  margin-right: 8px;
  font-size: 14px;
  min-width: 175px;
  text-align: right;
}

.c2 {
  border: solid 1px #C8CBCC;
  height: 28px;
  padding: 0 8px;
  border-radius: 4px;
  font-size: 14px;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor="pizza-party"
  >
    üçïüéâ
  </label>
  <input
    className="c2 c3"
    id="pizza-party"
    name="pizzaParty"
    placeholder={undefined}
    required={undefined}
    theme={
      Object {
        "colors": Object {
          "borderColor": "#C8CBCC",
          "destructive": "#CC1F36",
          "destructiveDark": "#B2121F",
          "destructiveDarker": "#990F14",
          "destructiveLight": "#FF667A",
          "destructiveLighter": "#FFB8C1",
          "destructiveText": "#FFFFFF",
          "linkColor": "#3973E5",
          "primary": "#64518A",
          "primaryDark": "#473764",
          "primaryDarker": "#2A1B60",
          "primaryLight": "#E5D9FF",
          "primaryLighter": "#F1EBFF",
          "primaryText": "#FFFFFF",
        },
        "shadows": Object {
          "large": "0 6px 12px rgba(0, 0, 0, 0.12)",
          "medium": "0 4px 8px rgba(0, 0, 0, 0.12)",
          "small": "0 2px 4px rgba(0, 0, 0, 0.08)",
          "xlarge": "0 8px 16px rgba(0, 0, 0, 0.12)",
        },
        "spacing": Object {
          "l": "1.25rem",
          "m": "1rem",
          "s": "0.5rem",
          "xl": "1.5rem",
          "xs": "0.25rem",
          "xxl": "1.875rem",
          "xxxl": "2.25rem",
          "xxxxl": "2.875rem",
        },
        "transitions": Object {
          "durationLong": 375,
          "durationShort": 200,
          "durationShorter": 150,
          "durationStandard": 300,
        },
      }
    }
    type="text"
  />
</div>
`;
